File: python/examples/Fade.py
import sys
sys.path.append('../')
import rgb1602
import time

#define REG_RED         0x04        // pwm2
#define REG_GREEN       0x03        // pwm1
#define REG_BLUE        0x02        // pwm0

def breath(color):
  for i in range(0,255,1):
    lcd.setPWM(color, i)
    time.sleep(0.005)
    
  time.sleep(0.5)
  for i in range(254,0,-1):
    lcd.setPWM(color, i)
    time.sleep(0.005)

  time.sleep(0.5)


lcd=rgb1602.RGB1602(16,2)                               #create LCD object,specify col and row
lcd.printout("fade demo")
lcd.setColorAll()


while True:
  breath(0x04)
  breath(0x03)
  breath(0x02)

File: cpp/examples/Fade.cpp
#include "../DFRobot_RGBLCD.h"

DFRobot_RGBLCD lcd(16, 2);

void breath(unsigned char color){
  for(int i=0; i<255; i++){
    lcd.setPWM(color, i);
    delay(5);
  }
  delay(500);
  
  for(int i=254; i>=0; i--){
    lcd.setPWM(color, i);
    delay(5);
  }
  delay(500);
}

int main(){
  lcd.init();
  lcd.print("fade demo");
  lcd.setColorAll();

  while (1) {
    breath(REG_RED);
    breath(REG_GREEN);
    breath(REG_BLUE);
  }
  return 0;
}
